src/timestamp.cpp

```diff
@@ -0,0 +1,53 @@
+#include "g3log/time.hpp"
+#include "g3log/timestamp.hpp"
+#include <sstream>
+
+
+namespace g3 {
+
+
+	Timestamp::Timestamp(std::string& defaultFormat) :
```

Comment:
Sure, will do :)

Quick update: I started implementing changes based on using timeval struct. This struct should be set by gettimeofday. I got curious about the second parameter of this function and clarified that it is depricated:
_"POSIX.1-2008 marks gettimeofday() as obsolete, recommending the use of clock_gettime(2) instead"_

So I looked at clock_gettime which uses timespec struct:
- It is very similar to timeval except that it uses fractional part at nanosec precision
- It part of time.h or ctime header which makes me think it is supported on Windows

I made changes (still very raw) to use timespec and so far it works as expected. Please let me know what you think ;)